src/Microsoft.AspNet.Mvc.ModelBinding/ModelStateDictionary.cs

```diff
@@ -369,6 +370,73 @@ public void SetModelValue([NotNull] string key, [NotNull] ValueProviderResult va
             GetModelStateForKey(key).Value = value;
         }
 
+        /// <summary>
+        /// Clears <see cref="ModelStateDictionary"/> entries for model in specified <see cref="ModelMetadata"/>.
+        /// </summary>
+        /// <param name="modelMetadata">The <see cref="ModelMetadata"/>.</param>
+        /// <param name="modelKey">The entry to clear. </param>
+        /// <param name="modelMetadataProvider">The <see cref="IModelMetadataProvider"/>.</param>
+        public void ClearModelStateDictionaryForModel(
```

Comment:
A. The original comment was for not depending on model metadata, are you saying for null or empty prefix, we can fallback to the model metadata and hence it should be passed here? If so I do not agree, this complicates the main use case of the API and IMHO it is far more clean to have a separate method. 
B. Clearing the dictionary entirely is not the intent or the goal for this fix.

- Lets chat offline if you disagree.